include/NovelRT.Interop/Utilities/NrtEvent.h

```diff
@@ -0,0 +1,46 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+#ifndef NOVELRT_INTEROP_UTILITIES_NRTEVENT_H
+#define NOVELRT_INTEROP_UTILITIES_NRTEVENT_H
+
+#include "../NrtTypedefs.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+    // Event<>
+    NrtUtilitiesEventHandle Nrt_Event_Create();
+
+    NrtResult Nrt_Event_AddEventHandler(NrtUtilitiesEventHandle event,
+                                        void (*handler)(void*),
+                                        void* context,
+                                        NrtAtom* outputEventHandlerId);
+
+    NrtResult Nrt_Event_RemoveEventHandler(NrtUtilitiesEventHandle event, NrtAtom eventHandlerId);
+
+    NrtResult Nrt_Event_Invoke(NrtUtilitiesEventHandle event);
+
+    NrtResult Nrt_Event_Destroy(NrtUtilitiesEventHandle event);
+
+    // Event<NovelRT::Timing::Timestamp>
+    NrtUtilitiesEventWithTimestampHandle Nrt_EventWithTimestamp_Create();
+
+    NrtResult Nrt_EventWithTimestamp_AddEventHandler(NrtUtilitiesEventWithTimestampHandle event,
+                                                     void (*handler)(NrtTimestamp, void*),
+                                                     void* context,
+                                                     NrtAtom* outputEventHandlerId);
+
+    NrtResult Nrt_EventWithTimestamp_RemoveEventHandler(NrtUtilitiesEventWithTimestampHandle event,
+                                                        NrtAtom eventHandlerId);
+
+    NrtResult Nrt_EventWithTimestamp_Invoke(NrtUtilitiesEventWithTimestampHandle event, NrtTimestamp timestamp);
+
+    NrtResult Nrt_EventWithTimestamp_Destroy(NrtUtilitiesEventWithTimestampHandle event);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif // NOVELRT_INTEROP_UTILITIES_NRTEVENT_H
```

Comment:
I'm not sure this is how we agreed to wrap events long term. CC @capnkenny do you remember what we said about this way back when?

Reply:
This looks a bit cleaner than what we initially anticipated since we were going to explicitly wrap each defined type as an `Nrt` opaque handle...

With that said, @jeuxjeux20  we'll need unit tests added to ensure that `Nrt_Event_*` and friends work properly as our original concern was that since it could be of type `T`, C obviously does not have a very good way of handling this (although your solution looks to do just that).

Reply:
Created #548